src/http/keys/keys.service.ts

```diff
@@ -1,163 +1,76 @@
-import { Inject, Injectable, LoggerService, NotFoundException } from '@nestjs/common';
+import { Inject, Injectable, LoggerService } from '@nestjs/common';
 import { LOGGER_PROVIDER } from '@lido-nestjs/logger';
 import { KeyListResponse, KeyWithModuleAddress } from './entities';
-import { ConfigService } from 'common/config';
+import { StakingRouterService } from 'staking-router-modules/staking-router.service';
 import { ELBlockSnapshot, KeyQuery } from 'http/common/entities';
-import { CuratedModuleService, STAKING_MODULE_TYPE } from 'staking-router-modules/';
-import { KeysUpdateService } from 'jobs/keys-update';
-import { httpExceptionTooEarlyResp } from 'http/common/entities/http-exceptions/too-early-resp';
+import { KeyField } from 'staking-router-modules/interfaces/filters';
+import { IsolationLevel } from '@mikro-orm/core';
+import { EntityManager } from '@mikro-orm/knex';
 
 @Injectable()
 export class KeysService {
   constructor(
     @Inject(LOGGER_PROVIDER) protected readonly logger: LoggerService,
-    protected curatedService: CuratedModuleService,
-    protected configService: ConfigService,
-    protected keysUpdateService: KeysUpdateService,
+    protected stakingRouterService: StakingRouterService,
+    protected readonly entityManager: EntityManager,
   ) {}
 
-  async get(filters: KeyQuery): Promise<any> {
-    const stakingModules = await this.keysUpdateService.getStakingModules();
+  async get(
+    filters: KeyQuery,
+  ): Promise<{ keysGenerators: AsyncGenerator<KeyWithModuleAddress>[]; meta: { elBlockSnapshot: ELBlockSnapshot } }> {
+    const { stakingModules, elBlockSnapshot } = await this.stakingRouterService.getStakingModulesAndMeta();
+    const keysGenerators: AsyncGenerator<KeyWithModuleAddress>[] = [];
 
-    if (stakingModules.length === 0) {
-      this.logger.warn('No staking modules in list. Maybe didnt fetched from SR yet');
-      throw httpExceptionTooEarlyResp();
-    }
+    for (const module of stakingModules) {
+      const moduleInstance = this.stakingRouterService.getStakingRouterModuleImpl(module.type);
 
-    // keys could be of type CuratedKey | CommunityKey
-    // const collectedKeys: KeyWithModuleAddress[][] = [];
-    let elBlockSnapshot: ELBlockSnapshot | null = null;
-    const { keysStream, meta } = await this.curatedService.getKeysWithMetaStream({
-      used: filters.used,
-      operatorIndex: filters.operatorIndex,
-    });
+      const fields: KeyField[] = ['key', 'depositSignature', 'operatorIndex', 'used', 'moduleAddress'];
```

Comment:
Strange implementation. Why just not fetch all keys from all modules and left join modules address

Reply:
Also maybe add FK from keys to operator and to module?